### ISSUE TITLE
Jekyll serve : wrong host for assets

### ISSUE THREAD

**Issue by jorismulliez** [envygeeks/jekyll-docker#211]
Hi,

I'm currently trying you image which seems to work very well.
My goal is to develop my website on my computer.

I'm new to docker and jekyll, so I probably missed something :)

- [x] I Am on Linux (mint)

## Description

I want to serve my jekyll website with the following command : 
```bash
JEKYLL_VERSION=3.7.3 docker run --rm --volume="$PWD:/srv/jekyll" -it jekyll/jekyll:3.7.3 jekyll serve
```
If I go to my container ip + port 4000 : http://172.17.0.2:4000/
It works, excepted for assets which are served from 0.0.0.0.
Example : 
```html
  <link rel="stylesheet" href="http://0.0.0.0:4000/assets/style.css">
```
So how can I tell jekyll to use 172.17.0.2 ?
I try this : 
```bash
JEKYLL_VERSION=3.7.3 docker run --rm   --volume="$PWD:/srv/jekyll"   -it jekyll/jekyll:3.7.3   jekyll serve -H 127.17.0.2
```
But I can't reach the website anymore...

And an other question : each time I launch the build or serve command through docker, I see a long list of dependencies fetched and installed (gem dependencies I suppose).
How to avoid each-time re-installation ?

In short: can you give instructions for beginners?

Thanks for your help,

**Comment by envygeeks** [comment]
Hi @jorismulliez.

There are directions in the README about how to speed up gem dependencies.  If it always outputs even with those directions that issue needs to be taken up with Bundler, we do offer caching of dependencies, and even check to make sure dependencies are matched *before* updating.

As far os the crux of this issue, I don't know what you are trying to achieve, or why you are trying to achieve it.  But I do know you should be using Dockers native port forwarding.  If it's because of the "0.0.0.0" that has no affect on the security of your computer, as that only affects the container, and then you route it externally, or internally via `docker -p`  You can find more information on how to port via https://docs.docker.com/v17.09/engine/userguide/networking/default_network/binding

**Comment by tonietto** [comment]
Hello, I was asked to deploy jekyll as a service in a docker swarm server and had the same difficulty as @jorismulliez:

> It works, excepted for assets which are served from 0.0.0.0.
> Example :
> ```
>  <link rel="stylesheet" href="http://0.0.0.0:4000/assets/style.css">
> ```
> So how can I tell jekyll to use 172.17.0.2 ?

That was causing `400` errors with assets in production. But it ended up to be a jekyll configuration and nothing to do with this repo. This is how I fixed it:

- I added `host` to `_config.yml` (example: `https://www.example.com`)
- And provided `JEKYLL_ENV` as anything different from `development` (example: `production`)

That way the `host` value will not be overriden by jekyll's default (http://0.0.0.0:4000), resulting as expected: 

```
<link rel="stylesheet" href="https://www.example.com/assets/style.css">
```
